README.md

```diff
@@ -16,76 +17,100 @@ This is basically [GitHub Markdown](https://help.github.com/articles/github-flav
 * ```![Image Title](MyFilename/foo.png)``` Adds an image. Images should be in a directory named after the filename of the file referencing them (or the same directory as the file referencing them)
 
-It then converts the markdown to HTML and shoves it on the Espruino website. lovely.
+It then converts the Markdown to HTML and shoves it on the Espruino website. Lovely!
 
-JavaScript
-----------
+## JavaScript
 
 Any `.js` files in `examples` have a webpage created that uses the comments as markdown, and then adds the code as a code block right at the end.
 
 All other `.js` files are treated as modules. They are minified using Google's online closure compiler and the SIMPLE_OPTIMISATIONS flag. To get advanced optimisations, you must add the exact text `@compilation_level ADVANCED_OPTIMIZATIONS` into the comments at the head of the file.
 
-Building
--------
+## Build Requirements
 
-You just need node.js... For Linux (Debian/Ubuntu) you can do this with:
+### Step 1: Obtain Espruino Source Code
 
-```
-# Get newer node.js - you'll only need this on earlier Linux versions
-$ sudo apt-add-repository ppa:richarvey/nodejs 
-$ sudo apt-get update
+Checked out the `Espruino` source at the same same level as `EspruinoDocs` folder. Assuming you're in the `EspruinoDocs` folder...
 
-# Install node
-$ sudo apt-get install npm nodejs
 ```
-# Set Maximum Open Files
-On OSX, most likely the default amount of open files will be set too low.  This may cause
-an error during the build, like: "Error: EMFILE, too many open files 'tasks/File Converter.md'"
+$ cd ..
+$ git clone git@github.com:espruino/Espruino.git`
+$ cd EspruinoDocs
+```
 
-Make sure you have at least 1024 for the value of open files.
 
-```
-$ ulimit -n       # see current limit
-```
+### Step 2: Install Node.js® and npm
+
+If you have Node.js and npm installed skip this step.
+
+If you haven't got Node.js JavaScript runtime installed or the JavaScript Package managers installed  do so via the installation guides below.
+
+* [Windows Installation Guide](http://treehouse.github.io/installation-guides/windows/node-windows.html)
+* [Mac Installation Guide](http://treehouse.github.io/installation-guides/mac/node-mac.html)
+* [Linux Installation Guide](http://treehouse.github.io/installation-guides/linux/node-linux.html)
 
-Increase the limit:
+
+
+### Step 3: Install Required Node Modules
+
+In order to generate the documentation and view it you require several JavaScript packages. To install them issue the following command:
 
 ```
-$ ulimit -n 1024  # increase to 1024
+$ npm install
 ```
 
-# Install Required Modules
+This will install all JavaScript dependancies.
 
-For OSX, go to [http://nodejs.org/](http://nodejs.org/) and click "Install".
+## Build Process
 
-Finally you need to install a node module for highlighting:
+### First Time:
+
+For the first time you can run:
 
 ```
-$ npm install marked highlight.js acorn express tern --save
+$ ./build.sh
```

Comment:
This might not be entirely right. For instance if you want your modules to be updated, or the pinout for each board to be updated, you'll still need build.sh I think?